daemon/api_download.go

```diff
@@ -37,7 +37,8 @@ var snapDownloadCmd = &Command{
 
 // SnapDownloadAction is used to request a snap download
 type snapDownloadAction struct {
-	SnapName string `json:"snap-name,omitempty"`
+	SnapName string              `json:"snap-name,omitempty"`
+	Options  snapRevisionOptions `json:"options,omitempty"`
```

Comment:
this probably should be embedded, we don't have "options" in any other API

Reply:
Thanks! I embedded this now and refactored/updated the tests a bit.